### PR TITLE
Separate deploy examples

### PR DESCRIPTION
**Pull request by devinmarieb**
# Description
Do not merge until redirects are set up for the overview page (I will do this 7/27)

`/deployment-examples` is separated out into individual pages for each example:

- /deploy-to-artifactory
- /deploy-to-aws
- /deploy-to-azure-container-registry
- /deploy-to-capistrano
- /deploy-to-cloud-foundry
- /deploy-to-firebase
- /authorize-google-cloud-sdk
- /deploy-to-google-cloud-platform
- /deploy-to-heroku
- /deploy-ios-applications
- /deploy-to-npm-registry
- /deploy-over-ssh
- /publish-packages-to-packagecloud

All other deployment example pages are now archived:

  - /deployment
  - /deployments
  - /deployment-examples
  - /deployment-integrations (became /deployment-overview as there are no longer integration examples on this page)

# Reasons
Separating out this content future proofs our deployment section so the examples page doesn't get exceedingly long, and we expect to add more examples in the future.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: https://circleci.com/docs/style/.

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

**Comment by devinmarieb** [comment]
@Mayuko-Sa This is the deployment PR for separating out the one deployment-examples page into separate pages for each individual example. This will likely get reviewed by docs and merged sometime in the middle or end of next week. Let me know if you have questions or if you'd like us to hold off on merging this till later for any reason 🙌

**Review by Mayuko-Sa — APPROVED** [review]

File: jekyll/_cci2/deploy-to-capistrano.adoc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+- Server v2.x
+---
+= Deploy to Capistrano
+:page-layout: classic-docs
+:page-liquid:
+:icons: font
+:toc: macro
+:toc-title:
+
+This pages proivdes a configuration for deployment to Capistrano. Once your project is set up to use Capistrano, you can run link:https://github.com/capistrano/capistrano/blob/master/README.md#command-line-usage[deployment commands] within your CircleCI job steps as required.
```
Comment (rosieyohannan) [inline jekyll/_cci2/deploy-to-capistrano.adoc:15]:
```suggestion
This page provides a configuration for deployment to Capistrano. Once your project is set up to use Capistrano, you can run link:https://github.com/capistrano/capistrano/blob/master/README.md#command-line-usage[deployment commands] within your CircleCI job steps as required.
```

File: jekyll/_cci2/deploy-to-cloud-foundry.adoc [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+- Server v2.x
+---
+= Deploy to Cloud Foundry
+:page-layout: classic-docs
+:page-liquid:
+:icons: font
+:toc: macro
+:toc-title:
+
+This pages proivdes a configuration for deployment to Cloud Foundry. CircleCI has developed a CloudFoundry Orb that you can use to simplify your configuration workflows. The Cloud Foundry page in the link:https://circleci.com/developer/orbs/orb/circleci/cloudfoundry[orbs registry] contains several different examples of how you can perform tasks with CloudFoundry, including the example below that shows how you can build and run blue green deployment in a single job - in this example `domain` will automatically be prefixed with `dark` and `live` for two subdomains to be specified. Validation steps would also need to be provided to allow the live deployment to go ahead.
```
Comment (rosieyohannan) [inline jekyll/_cci2/deploy-to-cloud-foundry.adoc:15]:
```suggestion
This pages proivdes a configuration for deployment to Cloud Foundry. CircleCI has developed a CloudFoundry orb that you can use to simplify your configuration workflows. The Cloud Foundry page in the link:https://circleci.com/developer/orbs/orb/circleci/cloudfoundry[orbs registry] contains several different examples of how you can perform tasks with CloudFoundry, including the example below that shows how you can build and run blue green deployment in a single job. In this example, `domain` will automatically be prefixed with `dark` and `live` for two subdomains to be specified. Validation steps would also need to be provided to allow the live deployment to go ahead.
```

File: jekyll/_cci2/deploy-to-cloud-foundry.adoc [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+          manifest: null
+          org: your-org
+          package: null
+          space: your-space
+          validate_steps:
+            # Call any orbs or custom commands that validate the health of deployed application before letting Green deploy/reroute proceed.
+            # For example,  hitting a /health endpoint with curl and making sure the dark URL returns a 200.
+```
+
+If you would like more detailed information about various CloudFoundry orb elements that you can use in your configuration workflows, refer to the link:https://circleci.com/developer/orbs/orb/circleci/cloudfoundry[CloudFoundry orb] page in the link:https://circleci.com/developer/orbs[CircleCI Orbs Registry].
```
Comment (rosieyohannan) [inline jekyll/_cci2/deploy-to-cloud-foundry.adoc:46]:
```suggestion
If you would like more detailed information about various CloudFoundry orb elements that you can use in your configuration, refer to the link:https://circleci.com/developer/orbs/orb/circleci/cloudfoundry[CloudFoundry orb] page in the link:https://circleci.com/developer/orbs[CircleCI Orbs Registry].
```

File: jekyll/_cci2/deploy-to-heroku.adoc [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        - build
+        - deploy:
+            requires:
+            - build # only run deploy-via-git job if the build job has completed
+            filters:
+            branches:
+                only: main # only run deploy-via-git job on main branch
+```
+
+**Note:** Heroku provides the option "Wait for CI to pass before deploy" under deploy / automatic deploys. See the link:https://devcenter.heroku.com/articles/github-integration#automatic-deploys[Heroku documentation] for details.
```
Comment (rosieyohannan) [inline jekyll/_cci2/deploy-to-heroku.adoc:82]:
```suggestion
NOTE: Heroku provides the option "Wait for CI to pass before deploy" under deploy / automatic deploys. See the link:https://devcenter.heroku.com/articles/github-integration#automatic-deploys[Heroku documentation] for details.
```

**Pull request merged**